src/x86/x86_model.py

```diff
@@ -635,30 +642,30 @@ def speculate_instruction(emulator: Uc, address, size, model) -> None:
                 reg_id = X86UnicornTargetDesc.reg_decode[reg]
                 reg_value = model.emulator.reg_read(reg_id)
                 source_taints.add(reg_value)
+            # print(f"Source taints:{source_taints}")
         
         # taint destination registers with new taints
         #   old taint can be overwritten
         for reg in reg_dest_operands:
             model.reg_taints[reg] = source_taints
+        # print(f"Dest taints:{model.reg_taints}")
             
-        
-
     @staticmethod
     def trace_mem_access(emulator, access, address, size, value, model) -> None:
         assert isinstance(model, X86UnicornVSPECUnknown)
         # print('memory access, curr observation:', model.curr_observation)
         if model.curr_observation:
             # print('speculative memory access, curr observation:', model.curr_observation)
             # do not access memory, just add memory access to observations
-            # print('current taints:', model.reg_taints)            
+            # print('current taints:', model.reg_taints)
             observation_list = list(model.curr_observation)
             observation_list.sort()
             observation_hash = hash(tuple(observation_list))
             # print('memory access with observation:', model.curr_observation)
             # observation_hash =  model.curr_observation.pop()
             # print('hash:', observation_hash)
-            model.tracer.observe_mem_access(access, observation_hash, size, 1, model)
```

Comment:
Passing an address  which is not mapped, to `observer_mem_access` causes an Unicorn exception